### PR TITLE
Ghc 8.8 and 8.10

### PR DESCRIPTION
**Pull request by Siprj**

### CODE REVIEW

File: src/Data/Unjson.hs [hunk 1]
```diff
@@ -1478,7 +1469,7 @@ render = P.render . renderDoc
 
 -- | Render only selected part of structure documentation. Path should
 -- point to a subtree, if it does not then Nothing is returned.
-renderForPath :: (Functor m, Monad m) => Path -> UnjsonDef a -> m String
+renderForPath :: (MonadFail m, Monad m) => Path -> UnjsonDef a -> m String
```
Comment (arybczak) [inline src/Data/Unjson.hs]: Isn't Monad redundant here? Same for below changes.

File: unjson.cabal [hunk 2]
```diff
@@ -15,8 +15,8 @@ copyright:           Scrive AB
 category:            Data
 build-type:          Simple
 extra-source-files:  README.md, CHANGELOG.md
-tested-with:         GHC == 7.8.4, GHC == 7.10.3, GHC == 8.0.2,
-                     GHC == 8.2.2, GHC == 8.4.4, GHC == 8.6.2
+tested-with:         GHC == 8.0.2, GHC == 8.2.2, GHC == 8.4.4, GHC == 8.6.5,
```
Comment (arybczak) [inline unjson.cabal]: Can you make it have the same format as hpqtypes?

File: unjson.cabal [hunk 3]
```diff
@@ -31,7 +31,7 @@ library
   exposed-modules:     Data.Unjson
   -- other-modules:
   -- other-extensions:
-  build-depends:       base >= 4.7 && < 4.13,
+  build-depends:       base >= 4.7 && < 4.15,
```
Comment (arybczak) [inline unjson.cabal]: `4.9` has to be the lower bound.

File: test/Test.hs [hunk 4]
```diff
@@ -212,13 +205,17 @@ test_tuple_parsing = "Tuple parsing" ~: do
 
   let Result (_ :: (Integer, Integer, Text.Text)) iss'' = parse unjsonDef json
   assertEqual "Issues in parsing"
-                [ Anchored (Path [PathElemIndex 0]) "expected Integer, encountered String"
-                , Anchored (Path [PathElemIndex 1]) "expected Integer, encountered String"
-                , Anchored (Path [PathElemIndex 2]) "expected Text, encountered Number"
-                ] iss''
+    [ Anchored (Path [PathElemIndex 0])
+        "parsing Integer failed, expected Number, but encountered String"
```
Comment (arybczak) [inline test/Test.hs:209]: Ah, I see old messages were inaccurate, nice.

**Review by arybczak — CHANGES_REQUESTED** [review]

**Review by arybczak — APPROVED** [review]
LGTM :+1: :tada:

**Comment by arybczak** [comment]
Before merging you can also update `lts` in `stack.yml` (it's part of #17) so that we can close both #17  and #15 as as far as I see this PR includes both changes.

**Comment by soareschen** [comment]
@arybczak has this been published to Hackage?

**Comment by arybczak** [comment]
It was now.

**Pull request merged**